src/silo/query_engine/exec_node/ndjson_sink.cpp

```diff
@@ -92,7 +98,9 @@ arrow::Status createGenerator(
    arrow::acero::ExecNode* input,
```

Comment:
The arrow docs warn that https://arrow.apache.org/docs/cpp/acero/overview.html#execplan, 
"ExecPlan is an internal concept. Users creating plans should be using Declaration objects. APIs for consuming and executing plans should abstract away the details of the underlying plan and not expose the object itself."

But I see we are creating an ExecPlan and passing it to `QueryPlan::makeQueryPlan` is this a problem?